packages/synchronize/lib/src/object.dart

```diff
@@ -0,0 +1,22 @@
+part of '../synchronize.dart';
+
+/// Wraps the actual data contained on each side.
+class SyncObject<T> {
+  // ignore: public_member_api_docs
+  SyncObject(
+    this.id,
+    this.data,
+  );
+
+  /// Id of the object.
+  final String id;
+
+  /// Actual data of the object, can be anything.
+  final T data;
+}
+
+// ignore: public_member_api_docs
+extension SyncObjectsFind<T> on List<SyncObject<T>> {
+  // ignore: public_member_api_docs
+  SyncObject<T>? find(final String id) => firstWhereOrNull((final object) => object.id == id);
```

Comment:
```suggestion
  SyncObject<T>? tryFind(final String id) => firstWhereOrNull((final object) => object.id == id);
```

We only use find when the result is non nullable